src/service.ts

```diff
@@ -331,6 +332,37 @@ export class Service {
     }
     return uri;
   }
+
+  static downloadProject(project: Project, uri?: string) {
+    const zipFile: JSZip = new JSZip();
+    let zipName: string = "download.zip";
+    if (!isUndefined(uri)) {
+      zipName = `${uri}.zip`;
+    }
+    function defineZip(file: File, folder?: any) {
+      if (file instanceof Directory) {
+        let folder: any;
+        if (!(file instanceof Project)) {
+          folder = zipFile.folder(file.name);
+        }
+        file.mapEachFile((file: File) => defineZip(file, folder));
+      } else {
+        if (!isUndefined(folder)) {
+          folder.file(file.name, file.data);
+        } else {
+          zipFile.file(file.name, file.data);
+        }
+      }
+    }
+    defineZip(project);
+    zipFile.generateAsync({type: "base64"}).then(function(base64: any) {
+      const link = document.createElement("a");
+      link.download = zipName;
+      link.href = "data:application/zip;name=test.zip;base64," + base64;
```

Comment:
there is can be a limit on size of the URL. I recommend to use ['blob' type](https://stuk.github.io/jszip/documentation/api_jszip/generate_async.html#type-option) with generateAsync

Reply:
It will need FileSaver.js to provide zip as a download to the user. So basically it will add up more size in the bundle. Is there a way to provide zip through blob(without adding up more libs)?

Reply:
Sorry, I don't understand relation between FileSaver.js. URL.createObjectURL gives blob to be use in a web app.